### PR TITLE
 Add Org Column & drops Owner column from Catalog Table

### PR DESCRIPTION
**Pull request by SM43**
# Changes
Adds org column which would be org name of the catalog repo.
Owners will be added later as a array of user who will be authorized
to sync the catalog with Hub.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/test/fixtures/catalogs.yaml [hunk 1]
```diff
@@ -13,20 +13,20 @@
 # limitations under the License.
 
 - id: 1
-  name: Tekton
+  name: catalog-official
```
Comment (PuneetPunamiya) [inline api/test/fixtures/catalogs.yaml:16]: Should it be just `name: official` ?

File: api/cmd/db/migration.go [hunk 2]
```diff
@@ -33,6 +33,49 @@ func Migrate(api *app.APIConfig) error {
...
+						log.Error(err)
+						return err
+					}
+					if err := tx.Model(&model.Catalog{}).
+						AddUniqueIndex("idx_name_org_code", "name", "org").Error; err != nil {
+						log.Error(err)
+						return err
+					}
+
+					catalogQuery := `ALTER TABLE catalogs
```
Comment (sthaha) [inline api/cmd/db/migration.go:57]: wouldn't automigrate create these?

File: api/pkg/db/model/model.go [hunk 3]
```diff
@@ -35,13 +35,13 @@ type (
...
-		Name       string
-		Type       string
-		URL        string
-		Owner      string
-		ContextDir string
+		Name       string `gorm:"unique_index:idx_name_org_code"`
+		Org        string `gorm:"unique_index:idx_name_org_code"`
+		Type       string `gorm:"not null;default:null"`
+		URL        string `gorm:"not null;default:null"`
+		ContextDir string `gorm:"not null;default:null"`
```
Comment (sthaha) [inline api/pkg/db/model/model.go:42]: I am not sure if we need "not null". Also why is the default: null when there is a constraint called `not null` 🤔

File: config.yaml [hunk 4]
```diff
@@ -31,9 +31,9 @@ categories:
...
 
 catalogs:
-  - name: tektoncd
+  - name: catalog
+    org: tektoncd
     type: community
-    owner: tektoncd
     url: https://github.com/tektoncd/catalog
     revision: master
     contextDir: /
```
Comment (sthaha) [inline config.yaml:39]: this is supposed to be optional and treated as "" by default.

File: api/cmd/db/migration.go (hunk 2 above)
Comment (SM43) [reply api/cmd/db/migration.go:57]: automigrate does not update constraints
 http://gorm.io/docs/migration.html

File: config.yaml (hunk 4 above)
Comment (SM43) [reply config.yaml:39]: Yeah. I think we should remove this field from here and at the bottom of config file mention all fields their explanation

File: api/pkg/db/model/model.go (hunk 3 above)
Comment (SM43) [reply api/pkg/db/model/model.go:42]: `not null` is applied on table but when we create a record using `db.Create` if a field is not mentioned in struct, gorm adds 
`""` by default.
 So, for example if `url` is not defined then ideally it should return error but as gorm add `""`, it does not return error. due to which we add default as null and expects some value.

File: config.yaml (hunk 4 above)
Comment (sthaha) [reply config.yaml:39]: +1

**Pull request merged**